<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8" />
  <title>Home | Your Name</title>
  <link rel="stylesheet" href="style.css" />
</head>
<body>

<div class="sidebar">
  <div class="sidebar-links">
    <a href="index.html">Home</a>

    <div class="subsection">
      <a href="Lectures/Lecture1_WGS.pdf">Lecture 1</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1</div>
      <a href="Practicals/module1.html">Module 1</a>
      <a href="Practicals/module2.html">Module 2</a>
    </div>
    
    <div class="subsection">
      <a href="Lectures/lecture2_Variants.pdf">Lecture 1</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2</div>
      <a href="Practicals/module1.html">Module 1</a>
      <a href="Practicals/module2.html">Module 2</a>
    </div>

  </div>
</div>

  <div class="content">
    <h2>Genomic Analysis and Phylodynamics Workshop</h2>
    <img src="Pictures/Sequence_trees.png" alt="MainFigure" class="content-image">
    <h3>Pre-workshop Instructions</h3> 
    <p>Please click the link below for instructions to download and install all data and software required for this course:</p>
    <a href="pre-workshop_instructions.html">Pre-workshop Instructions</a>
    </p> 
    <br>
   <h3>Schedule - National Taiwan University 2025</h3>

<table>
  <thead>
    <tr>
      <th>Date</th>
      <th>Time</th>
      <th>Session</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><strong>Saturday 3rd May</strong></td>
      <td>9:30–10:30</td>
      <td><strong>Lecture 1: Introduction and Key Concepts</strong></td>
      <td>• Course outline<br>• Introduction to next generation sequencing and genomic epidemiology</td>
    </tr>
    <tr>
      <td></td>
      <td>10:40–12:40</td>
      <td><a href="Practicals/Obtaining_sequencing_data.md"><strong>Practical Session 1: Whole Genome Sequence Data Analysis</strong></a></td>
      <td>• Obtaining sequencing data<br>• Data manipulation and QC<br>• Reference-based mapping and de novo assembly</td>
    </tr>
    <tr>
      <td></td>
      <td>12:40–13:40</td>
      <td><strong>Lunch Break</strong></td>
      <td></td>
    </tr>
    <tr>
      <td></td>
      <td>13:40–14:15</td>
      <td><a href="Practicals/Obtaining_sequencing_data.md"><strong>Practical Session 1 (cont.): Whole Genome Sequence Data Analysis</strong></a></td>
      <td>• Catch-up, overview, QA</td>
    </tr>
    <tr>
      <td></td>
      <td>14:15–15:15</td>
      <td><strong>Lecture 2: Variant Detection and Phylogenetic Trees</strong></td>
      <td>• What is a variant? How do we call variants?<br>• Variant calling software and QC<br>• What are phylogenetic trees?<br>• Types of phylogenies, phylogenetic uncertainty (bootstrapping etc.)</td>
    </tr>
    <tr>
      <td></td>
      <td>15:30–18:00</td>
      <td><a href="Practicals/VariantCall.md"><strong>Practical Session 2: Variant Calling and Maximum Likelihood Trees</strong></a></td>
      <td>• Variant calling<br>• SNP filtering and QC<br>• Building SNP matrices<br>• Aligning consensus sequences<br>• Producing ML trees</td>
    </tr>
    <tr>
      <td><strong>Sunday 4th May</strong></td>
      <td>9:30–11:00</td>
      <td><strong>Lecture 3: Practical Applications of WGS and Phylogenetics</strong></td>
      <td>• Species identification<br>• Resistance and plasmid profiling<br>• Transmission<br>• Applications in real-world datasets</td>
    </tr>
    <tr>
      <td></td>
      <td>11:00–12:00</td>
      <td><a href="Practicals/Beast.md"><strong>Practical Session 3: Timed Phylogenetic Trees</strong></a></td>
      <td>• One-step timed phylogenetic tree with BEAST2<br>• Two-step timed phylogenies using ML + Bayesian frameworks</td>
    </tr>
    <tr>
      <td></td>
      <td>12:00–13:00</td>
      <td><strong>Lunch Break</strong></td>
      <td></td>
    </tr>
    <tr>
      <td></td>
      <td>13:00–14:30</td>
      <td><a href="Practicals/Beast.md"><strong>Practical Session 3 (cont.): Timed Phylogenetic Trees</strong></a></td>
      <td>• (cont.) One-step timed phylogenetic tree<br>• Two-step timed phylogenies using ML + Bayesian frameworks</td>
    </tr>
    <tr>
      <td></td>
      <td>14:45–16:45</td>
      <td><strong>Practical Session 4: Transmission and Profiling</strong></td>
      <td>• Identifying species, serotypes, and lineages from WGS<br>• Inferring transmission networks/clusters</td>
    </tr>
    <tr>
      <td></td>
      <td>16:45–17:00</td>
      <td><strong>Closing Remarks - Short Course</strong></td>
      <td>• Short course summary and feedback collection</td>
    </tr>
    <tr>
      <td><strong>Monday 5th May (Advanced)</strong></td>
      <td>9:00–10:45</td>
      <td><strong>Lecture 4: Advanced Applications of WGS</strong></td>
      <td>• Phylogeography and phylodynamics<br>• Recombination<br>• Average Nucleotide Identity (ANI)<br>• Mixed infection<br>• Fitness and selection</td>
    </tr>
    <tr>
      <td></td>
      <td>11:00–12:00</td>
      <td><strong>Practical Session 5: Mixed Infection, Recombination and ANI</strong></td>
      <td>• Identifying mixed infection<br>• Calculating ANI<br>• Testing for recombination</td>
    </tr>
    <tr>
      <td></td>
      <td>12:00–13:00</td>
      <td><strong>Lunch Break</strong></td>
      <td></td>
    </tr>
    <tr>
      <td></td>
      <td>13:00–14:00</td>
      <td><strong>Practical Session 5 (cont.): Mixed infection, Recombination and ANI</strong></td>
      <td>• (cont.) Identifying mixed infection<br>• Calculating ANI<br>• Testing for recombination</td>
    </tr>
    <tr>
      <td></td>
      <td>14:15–15:30</td>
      <td><strong>Practical Session 6: Phylogeography and Phylodynamics</strong></td>
      <td>• Phylogeography (ancestral state reconstruction) in R and BEAST2<br>• Phylodynamic analysis using a phylogeny (LBI, Skygrowth)</td>
    </tr>
    <tr>
      <td><strong>Tuesday 6th May (Advanced)</strong></td>
      <td>9:00–12:00</td>
      <td><strong>Practical Session 7: Fitness and Selection</strong></td>
      <td>• Advanced phylodynamic analysis in BEAST2 (strain fitness, Bayesian skyline, multi-state birth-death)<br>• Site-specific selection (homoplasy, dN/dS)<br>• GWAS</td>
    </tr>
    <tr>
      <td></td>
      <td>12:00–12:30</td>
      <td><strong>Closing Remarks - Advanced Course</strong></td>
      <td>• Full course summary and feedback collection</td>
    </tr>
  </tbody>
</table>

    <br>
    <p>If you have any questions, please <a href="mailto:ucbtsob@ucl.ac.uk">get in touch.</a></p>
    <br><br><br>
    <footer>
      © 2025 Ben Sobkowiak. Built using GitHub Pages.
    </footer>
  </div>

</body>
</html>
