<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8" />
  <title>Home | Your Name</title>
  <link rel="stylesheet" href="style.css" />
</head>
<body>

<div class="sidebar">
  <div class="sidebar-links">
    <a href="index.html">Home</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="Lectures/Lecture1_WGS.pdf">Lecture 1: Introduction and Key Concepts</a>
      <a href="Lectures/Lecture2_Variants.pdf">Lecture 2: Variant Detection and Phylogenetic Trees</a>
      <a href="Lectures/Lecture3_Phylodynamics.pdf.pdf">Lecture 3: Practical Applications of WGS and Phylogenetics</a>
      <a href="Lectures/Lecture4_Advanced.pdf">Lecture 4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="Practicals/Treetime.html">ML timed trees</a>
      <a href="Practicals/BactDating.html">Bayesian trees (Bactdaing)</a>
      <a href="Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="Practicals/Recombination.html">Detecting recombination</a>
      <a href="Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="Practicals/dnds.html">dN/dS</a>
      <a href="Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>

  <div class="content">
    <h2>Genomic Analysis and Phylodynamics Workshop</h2>
    <img src="Pictures/Sequence_trees.png" alt="MainFigure" class="content-image">
    <h3>Pre-workshop Instructions</h3> 
    <p>Please click the link below for instructions to download and install all data and software required for this course:</p>
    <a href="pre-workshop_instructions.html">Pre-workshop Instructions</a>
    </p> 
    <br>
   <h3>Schedule - National Taiwan University 2025</h3>

<table>
  <thead>
    <tr>
      <th>Date</th>
      <th>Time</th>
      <th>Session</th>
      <th>Modules</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><strong>Saturday 3rd May</strong></td>
      <td>9:30–10:30</td>
      <td><a href="Lectures/Lecture1_WGS.pdf"><strong>Lecture 1: Introduction and Key Concepts</strong></td>
      <td>• Course outline<br>• Introduction to next generation sequencing and genomic epidemiology</td>
    </tr>
    <tr>
      <td></td>
      <td>10:40–12:40</td>
      <td><a href="Practicals/Obtaining_sequencing_data.md"><strong>Practical Session 1: Whole Genome Sequence Data Analysis</strong></a></td>
      <td>• Obtaining sequencing data<br>• Data manipulation and QC<br>• Reference-based mapping and de novo assembly</td>
    </tr>
    <tr>
      <td></td>
      <td>12:40–13:40</td>
      <td><strong>Lunch Break</strong></td>
      <td></td>
    </tr>
    <tr>
      <td></td>
      <td>13:40–14:15</td>
      <td><a href="Practicals/Obtaining_sequencing_data.md"><strong>Practical Session 1 (cont.): Whole Genome Sequence Data Analysis</strong></a></td>
      <td>• Catch-up, overview, QA</td>
    </tr>
    <tr>
      <td></td>
      <td>14:15–15:15</td>
      <td><strong>Lecture 2: Variant Detection and Phylogenetic Trees</strong></td>
      <td>• What is a variant? How do we call variants?<br>• Variant calling software and QC<br>• What are phylogenetic trees?<br>• Types of phylogenies, phylogenetic uncertainty (bootstrapping etc.)</td>
    </tr>
    <tr>
      <td></td>
      <td>15:30–18:00</td>
      <td><a href="Practicals/VariantCall.md"><strong>Practical Session 2: Variant Calling and Maximum Likelihood Trees</strong></a></td>
      <td>• Variant calling<br>• SNP filtering and QC<br>• Building SNP matrices<br>• Aligning consensus sequences<br>• Producing ML trees</td>
    </tr>
    <tr>
      <td><strong>Sunday 4th May</strong></td>
      <td>9:30–11:00</td>
      <td><strong>Lecture 3: Practical Applications of WGS and Phylogenetics</strong></td>
      <td>• Species identification<br>• Resistance and plasmid profiling<br>• Transmission<br>• Applications in real-world datasets</td>
    </tr>
    <tr>
      <td></td>
      <td>11:00–12:00</td>
      <td><a href="Practicals/Beast.md"><strong>Practical Session 3: Timed Phylogenetic Trees</strong></a></td>
      <td>• One-step timed phylogenetic tree with BEAST2<br>• Two-step timed phylogenies using ML + Bayesian frameworks</td>
    </tr>
    <tr>
      <td></td>
      <td>12:00–13:00</td>
      <td><strong>Lunch Break</strong></td>
      <td></td>
    </tr>
    <tr>
      <td></td>
      <td>13:00–14:30</td>
      <td><a href="Practicals/Beast.md"><strong>Practical Session 3 (cont.): Timed Phylogenetic Trees</strong></a></td>
      <td>• (cont.) One-step timed phylogenetic tree<br>• Two-step timed phylogenies using ML + Bayesian frameworks</td>
    </tr>
    <tr>
      <td></td>
      <td>14:45–16:45</td>
      <td><strong>Practical Session 4: Transmission and Profiling</strong></td>
      <td>• Identifying species, serotypes, and lineages from WGS<br>• Inferring transmission networks/clusters</td>
    </tr>
    <tr>
      <td></td>
      <td>16:45–17:00</td>
      <td><strong>Closing Remarks - Short Course</strong></td>
      <td>• Short course summary and feedback collection</td>
    </tr>
    <tr>
      <td><strong>Monday 5th May (Advanced)</strong></td>
      <td>9:00–10:45</td>
      <td><strong>Lecture 4: Advanced Applications of WGS</strong></td>
      <td>• Phylogeography and phylodynamics<br>• Recombination<br>• Average Nucleotide Identity (ANI)<br>• Mixed infection<br>• Fitness and selection</td>
    </tr>
    <tr>
      <td></td>
      <td>11:00–12:00</td>
      <td><strong>Practical Session 5: Mixed Infection, Recombination and ANI</strong></td>
      <td>• Identifying mixed infection<br>• Calculating ANI<br>• Testing for recombination</td>
    </tr>
    <tr>
      <td></td>
      <td>12:00–13:00</td>
      <td><strong>Lunch Break</strong></td>
      <td></td>
    </tr>
    <tr>
      <td></td>
      <td>13:00–14:00</td>
      <td><strong>Practical Session 5 (cont.): Mixed infection, Recombination and ANI</strong></td>
      <td>• (cont.) Identifying mixed infection<br>• Calculating ANI<br>• Testing for recombination</td>
    </tr>
    <tr>
      <td></td>
      <td>14:15–15:30</td>
      <td><strong>Practical Session 6: Phylogeography and Phylodynamics</strong></td>
      <td>• Phylogeography (ancestral state reconstruction)<br>• Phylodynamic analysis with BEAST2 (Skyline analysis)</td>
    </tr>
    <tr>
      <td><strong>Tuesday 6th May (Advanced)</strong></td>
      <td>9:00–12:00</td>
      <td><strong>Practical Session 7: Fitness and Selection</strong></td>
      <td>• Strain-specific fitness (LBI)<br>• Site-specific selection (homoplasy, dN/dS)<br>• GWAS</td>
    </tr>
    <tr>
      <td></td>
      <td>12:00–12:30</td>
      <td><strong>Closing Remarks - Advanced Course</strong></td>
      <td>• Full course summary and feedback collection</td>
    </tr>
  </tbody>
</table>

    <br>
    <p>If you have any questions, please <a href="mailto:ucbtsob@ucl.ac.uk">get in touch.</a></p>
    <br><br><br>
    <footer>
      © 2025 Ben Sobkowiak. Built using GitHub Pages.
    </footer>
  </div>

</body>
</html>
