<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Instructions</title>
  <link rel="stylesheet" href="style.css">
</head>
<body>
  <div class="content">
<h2>Pre-course Instructions</h2>

<p>
Please follow these instructions to download all programs and data for the course. You will require ~1GB of space to install all programs and data.
</p>

<p>
Please create a new folder in which you will keep all the course materials and scripts (e.g., something like <code>/Users/yourname/Documents/Workshop/</code> in Mac or Linux, or <code>D:/Workshop/</code> in Windows).
</p>

<br>

<h2>Data and Scripts</h2>

<p>
Please download all data and scripts required for this course by clicking the following link (Note, the file size is ~ MB) and saving all the data and scripts contained in a folder on your computer.
</p>

<p>
<a href="https://drive.google.com/drive/folders/1w3WG0jCj9BfB6aMB2Kq7vC2Yfx1ck6NB?usp=share_link">Download Data and Scripts</a>
</p>

<br><br>

<h2>Programs and software</h2>

<h3>Mac and Linux users</h3>

<p>The easiest method for downloading all the software required for the workshop is to first download Miniconda.</p>

<h4>Mac users, please follow these instructions:</h4>

<ol>
  <li>Open the Terminal.</li>
  <li>Enter the following code:</li>
</ol>

<pre><code># Download and install Miniconda
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh -O ~/miniconda.sh
bash ~/miniconda.sh -b -p $HOME/miniconda

# Follow prompts, then restart terminal
</code></pre>

<h4>Linux users, please follow these instructions:</h4>

<ol>
  <li>Open the Terminal.</li>
  <li>Enter the following code:</li>
</ol>

<pre><code># Download and install Miniconda
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
bash ~/miniconda.sh -b -p $HOME/miniconda

# Follow prompts, then restart terminal
</code></pre>

<h4>Both Mac and Linux users:</h4>

<p>Following the link to download the scripts above, you should have downloaded three scripts: <code>Install_programs.sh</code>, <code>workshop_env.yml</code>, and <code>install_r_github.R</code>. These are the three scripts required to download all the programs required for the workshop.</p>

<ol>
  <li>Open your terminal again.</li>
  <li>Navigate to the folder where you downloaded the scripts and data (e.g., <code>/Users/yourname/Documents/workshop/</code>):</li>
</ol>

<pre><code>cd /Users/yourname/Documents/workshop/
</code></pre>

<ol start="3">
  <li>Run the following command:</li>
</ol>

<pre><code>bash Install_programs.sh
</code></pre>

<p>
This may take up to 30 minutes, and you should now have a Conda environment called <code>bioinfo-workshop</code> and can run all the analysis from there.
</p>

<p>To activate the environment, run:</p>

<pre><code>source "$HOME/miniconda/etc/profile.d/conda.sh"
conda activate bioinfo-workshop
</code></pre>

<br>

<h3>Windows users</h3>

<p>For Windows users, it's a bit more involved to install Miniconda.</p>

<h4>First, install WSL (Windows Subsystem for Linux)</h4>

<ol>
  <li>Open PowerShell as Administrator and run:</li>
</ol>

<pre><code>wsl --install
</code></pre>

<p>This will:</p>
<ul>
  <li>Install WSL2</li>
  <li>Install Ubuntu (default)</li>
  <li>Set up necessary components</li>
</ul>

<p>Restart your computer when prompted.</p>

<h4>Set up Ubuntu in WSL</h4>

<p>
After reboot, Ubuntu should open automatically. If not, open the WSL app manually. Wait for installation to complete, then choose a username and password when prompted. This opens a Linux terminal (Ubuntu shell).
</p>

<h4>Install Miniconda inside WSL</h4>

<pre><code>wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
bash miniconda.sh
</code></pre>

<p>Accept the license agreement and confirm the default install location (e.g., <code>/home/yourname/miniconda</code>).</p>

<p>You may get another prompt after this. Please type "no".</p>

<p>When installation finishes, close and reopen WSL, or run:</p>

<pre><code>source ~/.bashrc
</code></pre>

<p>Export Conda into your path with:</p>

<pre><code>export PATH=~/miniconda3/bin:$PATH
</code></pre>

<p>Check that Conda is available:</p>

<pre><code>conda --version
</code></pre>

<h4>Move Setup Scripts Into WSL</h4>

<p>Place the files (<code>Install_programs.sh</code>, <code>workshop_env.yml</code>, <code>install_r_github.R</code>) in a folder on Windows, e.g., <code>D:/Workshop/</code>.</p>

<p>In the Ubuntu terminal, navigate to that folder:</p>

<pre><code>cd D:/Workshop/
</code></pre>

<h4>Run the install script for Windows:</h4>

<pre><code>bash Install_programs_windows.sh
</code></pre>

<p>
After up to 30 minutes, you should have a Conda environment called <code>bioinfo-workshop</code> set up and ready.
</p>

<p>To activate it:</p>

<pre><code>
  source ~/miniconda3/etc/profile.d/conda.sh 
  conda activate bioinfo-workshop
</code></pre>

<br>

<h3>All users</h3>

<p>These programs must be installed directly from their websites:</p>

<ul>
  <li><a href="https://posit.co/download/rstudio-desktop/" target="_blank">RStudio</a></li>
  <li><a href="https://www.beast2.org/" target="_blank">BEAST2</a></li>
  <li><a href="http://tree.bio.ed.ac.uk/software/tracer/" target="_blank">Tracer</a></li>
  <li><a href="http://tree.bio.ed.ac.uk/software/figtree/" target="_blank">FigTree</a></li>
  <li><a href="https://ics.hutton.ac.uk/tablet/" target="_blank">Tablet</a></li>
  <li><a href="https://ormbunkar.se/aliview/" target="_blank">AliView</a></li>
</ul>

<br>
  Click <a href="https://bensobkowiak.github.io/BioinformaticsCourse/">here</a> to return to the course homepage.
    <br>
    <br>
  </div>
  </body>

<div class="sidebar">
  <div class="sidebar-links">
    <a href="index.html">Home</a>

    <a href="pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="Lectures/Lecture3_Phylodynamics.pdf.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="Practicals/Treetime.html">ML timed trees</a>
      <a href="Practicals/BactDating.html">Bayesian trees (Bactdaing)</a>
      <a href="Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="Practicals/Recombination.html">Detecting recombination</a>
      <a href="Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="Practicals/dnds.html">dN/dS</a>
      <a href="Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>

  
</html>
