<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>ANI</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (BactDating)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  
<h2>Average Nucleotide Identity</h2>
<p>The average nucleotide identity (ANI) measures the mean sequence identity of all shared orthologous regions between two genomes. It is widely used to define species boundaries in prokaryotes, with a typical cutoff of ~95â€“96% ANI indicating the same species. </p>
<p>ANI provides higher resolution than 16S rRNA comparisons and is computed using tools like fastANI or pyANI, though these can be quite complex to run. It also can give you a more formal measure how close an assembly is to known reference strains, and potentially whether they are different species.</p>
<p>Here we will use a custom function in Python (computeANI.py) to calculate the ANI from an assembly to different reference strains that have been provided to you.</p>
<p>The data we will use for this exercise are:</p>
<ul>
<li>
<p><strong>E.coli_K12.fasta</strong> - A FASTA file containing a reference strain of E. coli K-12 species.</p>
</li>
<li>
<p><strong>E.coli_O157H7.fasta</strong> - A FASTA file containing a reference strain of E. coli O157:H7 species.</p>
</li>
<li>
<p><strong>S.flexneri.fasta</strong> - A FASTA file containing a reference strain of Shigella flenxeri species.</p>
</li>
<li>
<p><strong>S.sonnei.fasta</strong> - A FASTA file containing a reference strain of Shigella sonnei species.</p>
</li>
<li>
<p><strong>H37Rv.fasta</strong> - A FASTA file containing a reference strain of Mycobacterium tuberculosis species.</p>
</li>
<li>
<p><strong>unknown.fasta</strong> - A FASTA file containing an unknown assembled bacterial genome.</p>
</li>
</ul>
<p><br></p>
<ol>
<li>Install the required dependencies into your conda environment.</li>
<pre><code class="language-bash">conda install -c bioconda blast
pip install biopython
</code></pre>
<p><em>These should already be downloaded but we want to make sure they are available to run the next command.</em></p>
<li>We can now calculate the ANI between our unknown assembly and each reference strain, one at time, with the following command (this will calculate ANI for the unknown assembly and the H37Rv <em>M. tuberculosis</em> strain):</li>
<pre><code class="language-bash">python computeANI.py Unknown.fasta H37Rv.fasta
</code></pre>
</ol>
<h4>Run the analysis again for each of the reference strains.</h4>
<h4><em>Questions</em></h4>
<ol>
<li>
<p>Which species is the unknown sample most likely to belong to?</p>
</li>
<li>
<p>Can you gain any more insights from this analysis about the relatedness among all the species tested?</p>
</li>
</ol>
<p><br></p>
<hr />
<h3>This is the end of the activities in practical session 5. Navigate back to the homepage for other activities <a href="../index.html">here</a>.</h3>
		
	</div>
  <script src="../script.js"></script>
</body>


  
</html>
