<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>BEAST</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (BactDating)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  
  
  <h2>Bayesian reconstruction of timed trees with BEAST2</h2>
<p>BEAST (Bayesian Evolutionary Analysis Sampling Trees) is a powerful software package widely used for inferring time-measured phylogenies from molecular sequence data. It employs a Bayesian framework to simultaneously estimate phylogenetic trees, evolutionary parameters, and divergence times, allowing researchers to integrate molecular sequence data with temporal information to reconstruct the evolutionary history of species and pathogens. BEAST is particularly valuable for timed phylogeny building due to its ability to model complex evolutionary processes, account for uncertainty in phylogenetic inference, and incorporate prior knowledge about the evolutionary rates and divergence times.</p>
<p>One of the key features that makes BEAST an important tool for timed phylogeny building is its flexibility in modeling evolutionary processes. BEAST allows users to specify sophisticated evolutionary models, including substitution models, molecular clock models, demographic models, and phylogeographic models, tailored to the specific characteristics of the data and the biological question of interest. Additionally, BEAST provides rigorous statistical methods, such as Bayesian Markov Chain Monte Carlo (MCMC) sampling, for estimating posterior distributions of model parameters, allowing for the quantification of uncertainty in the inferred phylogenies and divergence times. Overall, BEAST's combination of flexibility, sophistication, and statistical rigor makes it an indispensable tool for timed phylogeny building and advancing our understanding of evolutionary history across diverse taxa.</p>
<p>The data we will be using in this exercise are:</p>
<ul>
<li>
<p><strong>TB_Cluster.fasta</strong> – A FASTA alignment file of concatenated SNPs from 38 <em>M. tuberculosis</em> samples collected between 2005 - 2014 in British Columbia. These isolates all share a MIRU-VNTR type, suggesting they may be linked by transmission.</p>
</li>
<li>
<p><strong>TB_Cluster.txt</strong> – A text file with two columns, the name of the 38 <em>M. tuberculosis</em> samples and their collection dates.</p>
</li>
</ul>
<p><br></p>
<p>To run BEAST2, we first need to create XML file in BEAUti that can be read by the main tool - BEAST. BEAUti is included in the BEAST2 suite of software. BEAUti can also be run using the command line but here we will take you through the user interface. <br></p>
<h3>Open BEAUti:</h3>
<p><img src="../Pictures/beauti.jpg" alt="Description1" width="10%"/></p>
<p><br></p>
<h3>1. You will open the following screen. Click the + sign illustrated by the red box:</h3>
<p><img src="../Pictures/BEAST1.jpeg" alt="Description1" width="70%"/></p>
<p><br>   </p>
<h3>2. Navigate to the "TB_Cluster.fasta" file in your data folder and click open. You will then be prompted to choose the datatype, we will select "all are nucleotide":</h3>
<p>Other types of data can be included, such as binary character data, and multiple data sources for the same samples can be used together with different models applied. For example, you could use SNPs in the form of nucleotides and the presence/absence of indels as binary data to estimate the phylogeny, with different evolutionary models applied to each type of data.</p>
<p><img src="../Pictures/BEAST2.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>3. This will load the data into BEAUti. These are called 'partitions' and you can have multiple input datasets (e.g., SNPs and Indels). Here we are just using SNPs, the arrows below point to the key information on this screen:</h3>
<p><img src="../Pictures/BEAST3.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>4. Next, we will load in the date information. Click the 'Tip Dates' tab shown in the red box below:</h3>
<p><img src="../Pictures/BEAST4.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>5. Dates may be in various formats, and can also be contained in the sequence names or in a separate file. Here, we have are dates in the text file "TB_Cluster.txt" in the yyyy-M-dd format. So first we need to click the 'as dates with format' button and select the correct date format, then click 'Auto-configure':</h3>
<p><img src="../Pictures/BEAST5.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>6. Select 'read from file' and click the 'Browse' button to find the dates file in your data folder:</h3>
<p><img src="../Pictures/BEAST6.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>7. This will bring up the dates of each sample in the middle column and the distance to the most recent sequence in unit time:</h3>
<p><img src="../Pictures/BEAST7.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>8. Next, we will navigate to the 'Site Model' tab to select our nucleotide substitution model. Here we want to use GTR but other models are available:</h3>
<p><img src="../Pictures/BEAST8.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>9. The 'Clock Model' tab will allow us to pick a molecular clock model.</h3>
<p>This can be set to 'strict' if you believe that the molecular clock is constant across all branches of the tree or relaxed if you want a more flexible model that allows for clock rates to vary across branches of the tree. Here we can also set a prior value for our molecular clock in the box below. To read more about molecular clocks see <a href="https://www.sciencedirect.com/topics/immunology-and-microbiology/molecular-clock">here</a>:</p>
<p><img src="../Pictures/BEAST9.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>10. The last evolutionary model to choose is the Tree model, in the 'Priors' tab.This can be changed using the dropdown menu at the top.</h3>
<p>This parameter sets the population demographic model, which are used to infer historical changes in population size over time based on molecular sequence data. These models allow researchers to estimate parameters related to population dynamics, such as changes in effective population size, population growth rates, demographic bottlenecks, and migration rates. </p>
<p>Here we will set the population model as 'Coalescent Constant Model'. Please read more about the different population demographic models <a href="https://beast.community/tree_priors">here</a>.</p>
<p><img src="../Pictures/BEAST10.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>11. The last parameter to set is in "MCMC" tab. Here we can set the number of chains to run, which is the number of MCMC iterations to run the model for. Here we can set the number of MCMC chains to 1 million, this should take a few minutes to run.</h3>
<p>The number of chains to run the model for is dependent on the complexity of the data and the underlying models; longer runs with more likely lead to the MCMC chains to converge (reach an equilibrium):</p>
<p><img src="../Pictures/BEAST11.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>12. Finally, we need to save the XML file by selecting 'File' and 'Save As'. Save your file as "TB_Cluster.xml":</h3>
<p><img src="../Pictures/BEAST12.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>Now we have our XML file saved, we can open the BEAST program:</h3>
<p><img src="../Pictures/beast.jpg" alt="Description1" width="10%"/></p>
<p><br></p>
<h3>1. BEAST will look like this:</h3>
<p><img src="../Pictures/BEAST13.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>2. Load in your XML file into the Input file section and choose the number of threads to use:</h3>
<p><img src="../Pictures/BEAST14.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>3. Next, we can run BEAST by clicking 'Run'. This will start the BEAST run to output a posterior collection of trees that will produced at intervals through the MCMC iterations:</h3>
<p><img src="../Pictures/BEAST15.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>You should find that 3 files are created: TB_Cluster.log, TB_cluster.trees, and TB_Cluster.xml.state. We want to inspect how well our run has converged using the .log file. We can use 'Tracer' to analyze the output from BEAST.</h3>
<h3>Open Tracer:</h3>
<p><img src="../Pictures/Tracer.jpg" alt="Description1" width="10%"/></p>
<p><br></p>
<h3>1. You will open the following screen. Click the + in the red box and open the "TB_Cluster.log" file.</h3>
<p><img src="../Pictures/Tracer2.jpg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>2. This will bring up all of the posterior estimates of different parameters. Your results will look slightly different to these, and to each other, as BEAST is a stochastic program:</h3>
<p><img src="../Pictures/BEAST16.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>3. You can also view the traces of each parameter to see how well the MCMC has converged:</h3>
<p>Convergence refers to the property of an MCMC algorithm whereby it reaches a stationary distribution that accurately represents the posterior distribution of the model parameters. In simpler terms, it indicates that the chain has explored the parameter space sufficiently and is sampling from the true underlying distribution.</p>
<p><img src="../Pictures/BEAST17.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<p><em>Discussion: What parameters do you think have not converged. What may be the reason? View the log file provided in the Data folder "TB_Cluster_longer.log. How do these traces compare to your runs?</em></p>
<p><br></p>
<h3>Finally, we can create a single consensus tree using 'TreeAnnotator' from the posterior distribution of trees found in the "TB_Cluster.trees" file:</h3>
<h3>Open TreeAnnotator:</h3>
<p><img src="../Pictures/treeannotator.jpg" alt="Description1" width="10%"/></p>
<p><br></p>
<h3>1. Here will first select the percentage burn-in. This is initial portion of the MCMC chain where samples are discarded, removing any influence of the starting values on the estimation of posterior distributions. We will also open the TB_Cluster.trees file in the 'Input tree file' box, and name our single output tree in the 'Output file' box. We can name this "TB_Cluster.tree":</h3>
<p><img src="../Pictures/BEAST_25.jpg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>2. Then click 'Run' to estimate a single consensus tree from the posterior output from BEAST. This will be saved in the "TB_Cluster.tree" file:</h3>
<p><img src="../Pictures/BEAST19.jpg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>3. Open the "TB_Cluster.tree" file in FigTree:</h3>
<p><img src="../Pictures/BEAST20.jpg" alt="Description1" width="70%"/></p>
<p><br></p>
<h3>4. Explore the options on the bottom left to look at the estimated times of the nodes:</h3>
<p><img src="../Pictures/BEAST21.jpg" alt="Description1" width="70%"/></p>
<p><em>Question: How far in the past is the TMRCA (time to most recent common ancestor) of all the samples in our cluster?</em></p>
<p><br></p>
<h3>Further resources:</h3>
<h4>BEAST is very well documented with numerous basic and advanced tutorials. The excellent <a href="https://taming-the-beast.org/tutorials/Introduction-to-BEAST2/">'Taming the BEAST'</a> workshop has many different walkthoughs and tutorials to guide you through all aspects of runnnig BEAST.</h4>
<h3>This is the end of the activities in practical session 3. Navigate back to the homepage for other activities <a href="../index.html">here</a>.</h3>
  
		
	</div>
</body>


  
</html>
