<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Obtaining Data</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (BactDating)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  
  
  <h2>De novo (reference-free) assembly</h2>
<p>De novo assembly is the process of reconstructing genomes from sequencing read data without the use of a reference genome. It involves overlapping and merging reads to form longer contiguous sequences (contigs) that represent the genomic sequence. </p>
<p>This process is useful when no suitable reference genome is available or when the goal is to discover novel genomic features, such as structural variations, insertions, or new genes. It is also valuable for assembling organisms with high levels of recombination, as it allows reconstruction of unique genome arrangements without bias from a reference.</p>
<p>The goal is to assemble the read data into a single consensus sequence, though this is rarely achieved with short-read data and you will likely have multiple contigs. Long-read sequencing data is much more likely to </p>
<p>We will use a <em>de novo</em> assembly software called <a href="https://github.com/rrwick/Unicycler">Unicycler</a>. This program uses an assembly software called <a href="https://github.com/ablab/spades">SPAdes</a> to assemble the sequencing reads into contigs, polishes the assemblies to improve the accuracy, and where possible, will circularize genomes.</p>
<p>For this activity, we will use the following files found in your /data/ folder:</p>
<ul>
<li><strong>Kleb1_R1.fastq.gz</strong> and <strong>Kleb1_R2.fastq.gz</strong> â€“ Paired-end 150bp sequence data from <em>Klebsiella pneumoniae</em>, collected in Taiwan. These have been inspected and cleaned already.</li>
</ul>
<p><br></p>
<ol>
<li>First we can view the options in Unicycler by simply typing the following command in your terminal:</li>
</ol>
<pre><code class="language-bash">unicycler
</code></pre>
<p>This will display all options for Unicycler:</p>
<p><img src="../Pictures/Unicycler1.jpeg" alt="Description" width="60%"/></p>
<p><br></p>
<ol>
<li>We can then run the main command for Unicycler. This may take a little while to complete as it will test different k-mer lengths (see Lecture 1 for k-mer definition) to produce the optimal assembly. Perhaps run this command over the lunch break:</li>
</ol>
<pre><code class="language-bash">unicycler -1 Kleb1_R1.fastq.gz -2 Kleb1_R2.fastq.gz --keep 0 -t 2 -o Kleb1
</code></pre>
<p>Where the options specified are: </p>
<table>
<thead>
<tr>
<th>Option</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-1</td>
<td>The forward read of short-read paired end sequence data</td>
</tr>
<tr>
<td>-2</td>
<td>The reverse read of short-read paired end sequence data</td>
</tr>
<tr>
<td>--keep</td>
<td>The file to keep in the output directory.  0 = only keep final files: assembly (FASTA, GFA and log),</td>
</tr>
<tr>
<td>-t</td>
<td>The number of threads to use.</td>
</tr>
<tr>
<td>-o</td>
<td>The name of the output directory</td>
</tr>
</tbody>
</table>
<p>We will now have the de novo assembled contigs using the optimal k-mer length that Unicycler determines using some quality metrics (outlined on the Unicycler Github). These will be contained in the /Kleb1/ folder.</p>
<ol>
<li>Navigate to this folder and view the files:</li>
</ol>
<pre><code class="language-bash">cd Kleb1/
ls #this is the command to view files
</code></pre>
<p>These files include a FASTA file with the assembled contigs (assembly.fasta)</p>
<ol>
<li>View the fasta file in AliView:</li>
</ol>
<p><img src="../Pictures/denovo1.jpeg" alt="Description" width="60%"/></p>
<h4>Discuss the assembled contigs, do you think we have reconstructed the complete genome?</h4>
<p>Instead of inspecting the files manually, we can use software such as [QUAST}(https://github.com/ablab/quast) to calculate different metrics to assess our assemblies.</p>
<ol>
<li>Run QUAST with the following command to calculate assembly metrics and output the new files to a folder called QUAST/:</li>
</ol>
<pre><code class="language-bash">quast -o QUAST assembly.fasta
</code></pre>
<p>This will produce files in multiple formats that show the assembly metrics in the Kleb1/QUAST/ folder. We will view the 'report.txt; file.
<br></p>
<ol>
<li>Either open the report.txt file in your text editor, or view it in Terminal or WSL using the following command:</li>
</ol>
<pre><code class="language-bash">cd QUAST/
vi report.txt
</code></pre>
<p>The report.txt file look like this:</p>
<p><img src="../Pictures/denovo2.jpeg" alt="Description" width="60%"/></p>
<h4>Please discuss the output - 1. How many contigs do we have? 2. Given that the Klebsiella pneumoniae genome is ~5.5MB, do we think we have reconstructed the full genome?</h4>
<p>Some of the metrics are not as intuitive (e.g., N50, L50). We may not have time to discuss but these are important to determine the contiguity of the assembly. Please read <a href="https://en.wikipedia.org/wiki/N50,_L50,_and_related_statistics">here</a> for more details.</p>
<hr />
<h3>This is the end of the activities in practical session 1. Navigate back to the homepage for other activities <a href="../index.html">here</a>.</h3>
  
		
	</div>
</body>


  
</html>