<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Transmission</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (BactDating)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  
<h2>Genome Wide Association Studies with
treeWAS</h2>


<p>Genome Wide Association Studies (GWAS) can be used to find
significant genomic associations with a given trait or characteristic
(phenotype). We can analyze the entire genome of individuals to pinpoint
single nucleotide polymorphisms (SNPs) or other genetic markers (e.g.,
the presence or absence of a gene) that exhibit statistical associations
with particular phenotypes. A key consideration when running a GWAS is
to control for population structure as this may induce false positive
calls in where variants are identified due to shared ancestry rather
than because of the trait of interest.</p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30184106/">treeWAS</a>
is a package available in R that has been specifically developed to run
bacterial GWAS. This method can be very powerful for identifying SNPs in
bacterial genomics that are significantly associated with a given
phenotype. This phenotype can be a binary or continuous variable.</p>
<p>This practical will guide you through running treeWAS to perform a
GWAS using the 200 <em>M. tuberculosis</em> samples from Moldova from
the BactDating practical in session 3.</p>
<p><br></p>
<div id="first-we-will-set-up-the-packages-and-options" class="section level4">
<h4>1. First we will set up the packages and options:</h4>
<pre class="r"><code># Set packages and download treeWAS
require(devtools)
require(treeWAS)
options(stringsAsFactors = F)</code></pre>
<p><br></p>
</div>
<div id="to-run-a-gwas-with-treewas-you-will-need-a-genotype-sequence-file-a-phenotype-file-and-a-phylogeny.-first-we-will-first-load-the-tree-file.-this-will-be-the-un-timed-ml-phylogeny-tb_moldova.tree" class="section level4">
<h4>2. To run a GWAS with treeWAS, you will need a genotype (sequence)
file, a phenotype file, and a phylogeny. First we will first load the
tree file. This will be the un-timed ML phylogeny
“TB_Moldova.tree”:</h4>
<pre class="r"><code>tree &lt;- ape::read.tree(&quot;TB_Moldova.tree&quot;) </code></pre>
<p><br></p>
</div>
<div id="next-load-in-the-genotype-file-in-dnabin-format-of-from-ape.-this-will-be-the-multi-sequence-fasta-file.-we-need-to-convert-this-to-a-specific-format-for-treewas-using-the-dnabin2genind-function" class="section level4">
<h4>3. Next load in the genotype file in DNAbin format of from ‘ape’.
This will be the multi-sequence FASTA file. We need to convert this to a
specific format for treeWAS using the ‘DNAbin2genind’ function:</h4>
<pre class="r"><code>dna &lt;- read.dna(file = &quot;TB_Moldova.fasta&quot;, format = &quot;fasta&quot;)
genotype &lt;- DNAbin2genind(dna)@tab</code></pre>
<p><br></p>
</div>
<div id="we-will-use-the-phenotype-found-in-the-third-column-of-the-tb_moldova_metadata.csv-file-you-have-been-provided-with.-this-will-be-converted-to-a-named-vector-using-the-sequence-names.-this-is-a-binary-variable-with-1-presence-of-phenotype-0-absence-of-phenotype." class="section level4">
<h4>4. We will use the phenotype found in the third column of the
“TB_Moldova_metadata.csv” file you have been provided with. This will be
converted to a named vector using the sequence names. This is a binary
variable, with 1 = presence of phenotype, 0 = absence of phenotype.</h4>
<pre class="r"><code>phenotype &lt;- read.csv(file= &quot;TB_Moldova_metadata.csv&quot;)
phen &lt;- as.vector(unlist(phenotype[,3])) # The 
names(phen) &lt;- phenotype[,1]</code></pre>
<p><br></p>
</div>
<div id="the-final-step-is-to-run-treewas-we-can-also-set-a-random-seed-to-replicate-the-result." class="section level4">
<h4>4. The final step is to run treeWAS, we can also set a random seed
to replicate the result.</h4>
<p>You will get 7 plots after the run completes. There will be one plot
with the input tree with tips coloured by the phenotype, and 6 results
plots. These will be a manhatten plot and a bar graph for each of three
statistical measures to determine signficant associations, with a red
line illustrating the cutoff for significance with bonferroni
correction.</p>
<ol style="list-style-type: decimal">
<li>the “Terminal Score”, measures sample-wide association across the
leaves ofthe phylogenetic tree.</li>
<li>the “Simultaneous Score”, measures the degree of parallel change in
the phenotype and genotype across branches of the tree.</li>
<li>the “Subsequent Score”, measures the proportion of the tree in which
the genotype and phenotype co-exist.</li>
</ol>
<pre class="r"><code>out &lt;- treeWAS(snps = genotype,
               phen = phen,
               tree = tree,
               seed = 2345)</code></pre>
<p><br></p>
</div>
<div id="the-results-will-show-you-the-coordinates-in-the-genotype-file-that-have-been-found-to-be-significantly-associated-with-the-phenotype." class="section level4">
<h4>The results will show you the coordinates in the genotype file that
have been found to be significantly associated with the phenotype.</h4>
<p><br></p>
</div>
<div id="exercise-you-have-been-provided-with-an-index-file-tb_moldova_index.txt-which-will-enable-you-to-link-the-coordinates-in-the-genotype-file-to-their-position-in-the-tb-h37rv-reference-strain.-find-the-genes-that-harbour-the-significant-snps-what-do-you-think-the-phenotype-is-in-our-analysis" class="section level4">
<h4>Exercise: You have been provided with an index file
“TB_Moldova_index.txt” which will enable you to link the coordinates in
the genotype file to their position in the TB H37Rv reference strain.
Find the genes that harbour the significant SNPs, what do you think the
phenotype is in our analysis?</h4>
<p><em>Tip: Go back to the NCBI tutorial on the first day to see how to
view reference strains.</em></p>
</div>
<div id="this-is-the-end-of-practical-7.-navigate-back-to-the-homepage-here" class="section level3">
<h3>This is the end of practical 7. Navigate back to the homepage <a href="../Index.html">here</a></h3>
<p><br></p>
</div>
		
	</div>
  <script src="../script.js"></script>
</body>


  
</html>
