<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Transmission</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (BactDating)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  <h2>Mixed Infection</h2>
<p>Mixed infection, where more than one distinct strain of a pathogen is
present in a host at the same time, can be a relatively common
occurrence in bacterial and viral infection. In TB for example, up to
20% of clinical samples have been estimated to be mixed.</p>
<p>Identifying these complex infections can be important clinically as
these samples may be hetero-resistant and the minor frequency strains
may be transmitting. Importantly, failing to account for potential mixed
infections can lead to a single erroneous consensus sequence that,
causing issues with downstream phylogenetic and genomic analysis.</p>
<p>This practical will walk you through the main steps to identify mixed
infection from allele frequencies from VCF files produced from
short-read sequence data. We will you my tool, <a href="https://pubmed.ncbi.nlm.nih.gov/39871355/">MixInfect2</a>, which
is freely available in R from the <a href="https://github.com/bensobkowiak/MixInfect2/">GitHub</a>, or using
‘devtools’ as we will see in the practical. Please read the paper if you
would like more information about how the tool works.</p>
<p>We will the following data for this exercise:</p>
<ul>
<li><p>SNPs_filtered.vcf - The multi-sample VCF file produced in
practical session 2 that contains SNP variant calls from 3 <em>M.
tuberculosis</em> samples.</p></li>
<li><p>MaskedRegions.csv - A .CSV file with genomic features that we
want to mask from the mixed infection analysis as variation in these
genes can obscure the signal from mixed infection.</p></li>
</ul>
<p><br></p>
<div id="first-we-will-set-up-the-mixinfect2-function-from-the-mixinfect2.r-script-in-your-folder" class="section level4">
<h4>1. First we will set up the MixInfect2 function from the
MixInfect2.R script in your folder:</h4>
<pre class="r"><code># Set packages 
source(&quot;MixInfect2.R&quot;)</code></pre>
<p><br></p>
</div>
<div id="next-we-can-run-mixinfect2.-this-will-take-the-vcf-file-and-the-csv-file-with-masked-regions-as-input" class="section level4">
<h4>2. Next, we can run MixInfect2. This will take the VCF file and the
CSV file with masked regions as input</h4>
<pre class="r"><code>results &lt;- MixInfect2(&quot;SNPs_filtered.vcf&quot;,maskFile = &quot;MaskedRegions.csv&quot;,prefix = &quot;TB_SNPs&quot;)</code></pre>
</div>
<div id="this-will-save-the-results-of-mixinfect2-to-the-results-variable.-view-the-results" class="section level4">
<h4>3. This will save the results of MixInfect2 to the results variable.
View the results:</h4>
</div>
<div id="we-can-see-that-we-have-some-potential-mixed-infection-in-our-three-samples.-below-are-two-allele-frequency-plots-from-our-samples." class="section level4">
<h4>4. We can see that we have some potential mixed infection in our
three samples. Below are two allele frequency plots from our
samples.</h4>
</div>
<div id="discussion-which-of-our-samples-do-you-think-these-plots-may-belong-to" class="section level4">
<h4><em>Discussion: Which of our samples do you think these plots may
belong to?</em></h4>
<p>You can use these results to remove putative mixed infection from
your analysis, or you can potentially reconstruct the constituent
sequences of mixed infection using the reconstructConstituents.R script
in MixInfect2, though there are limitations in this. Please see the
associated paper for more detail on reconstructing mixed constituent
sequences.</p>
</div>
<div id="next-activity-recombination" class="section level3">
<h3>Next activity: <a href="Recombination.html">Recombination</a></h3>
<p><br></p>
</div>
		
	</div>
  <script src="../script.js"></script>
</body>


  
</html>