<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>ML Trees</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (BactDating)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  
  
  <h2>Maximum Likelihood phylogenetic trees</h2>
<p>Once you have a sequence alignment, it is now possible to construct a phylogenetic tree. Approaches for building phylogenies can varying in complexity, from a relatively simple approach in Neighbour Joining trees that builds trees based on pairwise distances, to Maximum Likelihood and Bayesian methods. </p>
<p>These are more complex and computationally intensive, and incorporate statistical models to infer the most probable evolutionary scenarios while considering branch lengths and substitution rates and models of evolution, which offers a nuanced depiction of evolutionary processes.</p>
<p>For this exercise, you will be build a ML tree using the aligned and masked SARS-CoV-2 B.1.1.7 variant alignment we produced in the last exercise - <strong>"Taiwan_B117_aligned_masked.fasta"</strong>. Here we are using the full genome of SARS-CoV-2 but your input file could equally be any aligned sequence or partial sequence, for example an alignment of concatenated SNPs or an alignment of single or multiple genes in isolation. </p>
<p><em>Keep in mind though that if you are using a partial genome sequence then branch lengths and substitution rates will be scaled to the length of the input sequence rather than the full genome.</em></p>
<p>These ML trees can take time to build, depending on the number of threads you assign and the number and length of the sequences, each tree can take hours to complete. We have provided you the results in the data folder but try runnning these on your machines in your own time.</p>
<br>
	  <p><strong>Building ML phylogenies with IQtree</strong></p>
<p>The ML tree will build the phylogeny using nucleotide substitution models, which are mathematical representations used in phylogenetics to describe the rates and patterns of substitutions among nucleotides over evolutionary time. We touched in these models in the lecture, but more information is available <a href="https://www.sciencedirect.com/topics/medicine-and-dentistry/substitution-model">here</a>.</p>
<ol>
<li>We first want to perform a model test in IQtree to determine the optimal substitution model for our sequence data. More information about the options for this can be found <a href="http://www.iqtree.org/doc/Tutorial">here</a>.</li>
<pre><code class="language-bash">iqtree -s Taiwan_B117_aligned_masked.fasta -m TEST -T 2
</code></pre>
<p>This will create a zipped file called "Taiwan_B117_aligned_masked.fasta.model.gz" with the results of the model test.</p>
<li>Unzip it and open it using a text viewer. We have provided you with this result in the data folder.</li>
<p><em>Question: What is the best substitution model identified in this run?</em></p>
<p><br></p>
<li>Next we can build a ML tree using this optimal substitution model after the (-m) option. We can also produce a tree with branch and node support (how confident we are with the branch lengths and node placement) by running the command by bootstrapping with the -B option following by an integer, and an assessment of the branch support using -alrt following by an integer. More information on bootstrapping and branch support is available <a href="http://www.iqtree.org/doc/Tutorial">here</a>:</li>
<pre><code class="language-bash">iqtree -s Taiwan_B117_aligned_masked.fasta -m model --prefix Taiwan_B117_MLtree -alrt 1000 -B 1000 -keep-ident -T 2
</code></pre>
<p><em>*Replace 'model' with the name of the best performing model from the previous model test result.</em></p>
<p>The other options called in this command are: <br></p>
<ul>
<li>--prefix = The prefix to use for all output files. <br></li>
<li>-keep-ident = As standard, IQtree will only keep one identical sequence, add this option to keep all sequences.</li>
<li>-T = The number of threads</li>
</ul>
<p>There will be 9 output files created with this run, the important file will be the "Taiwan_B117_MLtree.treefile", which contains the best tree found by ML with the support values added.</p>
<p>Now we can view the "Taiwan_B117_MLtree.treefile". </p>
<li>
<p>Please open FigTree.</p>
</li>
<li>
<p>Go to File &gt; Open and select "Taiwan_B117_MLtree.treefile".</p>
</li>
<li>
<p>Click 'OK' on the next screen. Now you should see the tree.</p>
<p><img src="../Pictures/FigTree2.jpeg" alt="Description1" width="60%"/></p>
</li>
<li>
<p>It probably looks a bit messy due to the size of the tip labels (sequence names). We can reduce the size of the font to make the tree clearer by navigating to this panel on the right, clicking the arrow next to "Tip Labels" and reducing the font size:</p>
<p><img src="../Pictures/FigTree1.jpeg" alt="Description1" width="15%"/></p>
</li>
<li>
<p>Explore some other options to change the shape of the tree and bring up different labels.</p>
</li>
</ol>
<p><br></p>
<p>For more information on runnning IQtree, click <a href="http://www.iqtree.org/doc/Tutorial">here</a>.</p>
<h3>This is the end of the activities in practical session 2. Navigate back to the homepage for other activities <a href="../index.html">here</a>.</h3>
  
		
	</div>
</body>


  
</html>
