<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Obtaining Data</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (BactDating)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
		
		<h2>Obtaining sequence data</h2>
<p>While you will be provided with all the data you need for this course, normally you would either receive data directly from the sequencer/sequencing facility, from collaborators that hold the data (perhaps through a secure link), or download data from a number of publicly accessible databases.</p>
<p>For the first activity, we will introduce three of these databases and take you through how to search for and download sequences. The location and format of the sequence data will depend on the taxa of interest and whether you require the raw sequence data or partially or fully assembled sequences. </p>
<h3><a href="https://www.ncbi.nlm.nih.gov">National Center for Biotechnology Information (NCBI)</a></h3>
<p>NCBI provides access to a wide range of databases and tools, including the GenBank nucleotide database and SRA, the Sequence Read Archive. Genbank is great resource for accessing curated and well-characterized reference strains, providing a standardized basis for comparison and analysis across experiments, along with providing links to raw sequence data found in ENA. Most commonly used reference sequences will be referenced in publications with a GenBank accession number (e.g., NC_000962.3 for the <em>Mycobacterium tuberculosis</em> H37Rv reference strain.</p>
<p>Here we will go through how to download a complete assembly in the FASTA format, such as a reference strain:</p>
<h4>NCBI Genbank - downloading complete assemblies</h4>
<ol>
<li>
<p>Click the link to NCBI above.</p>
</li>
<li>
<p>You will be sent to a page (below): </p>
<p><img src="../Pictures/NCBI_1.jpeg" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>Type "NC_000962_3" into the search bar. This is the accession number of H37Rv.</p>
<p><img src="../Pictures/NCBI_2.jpeg" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<p>This will bring up the following result, click on the name 'Mycobacterium tuberculosis H37Rv, complete genome':</p>
<p><img src="../Pictures/NCBI_3.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<p>You will be directed to the result page for that genome:</p>
<p><img src="../Pictures/NCBI_4.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<li>
<p>To download the complete genome in a FASTA format file (a simple text-based format for representing nucleotide sequences), click "send to", choose "Complete record", "Choose Destination - File", and "FASTA" in the "format" dropdown menu. Then click "Create File"</p>
<p><img src="../Pictures/NCBI_5.jpeg" alt="Description1" width="70%"/></p>
</li>
</ol>
<p>This will download the full sequence in the FASTA format.</p>
<h4>NCBI SRA - downloading raw sequencing data</h4>
<p>Newly sequenced genomic data that is being reported in a publication will be submitted to SRA (or ENA below) and you can find the corresponding project ID number in the paper (often in a "Data availability" section). There will be a number like "Project ID ERP000436". You can download directly from the website, or use the SRA-toolkit to download many sequences. A good tutorial for using the SRA-toolkit is found [here]((https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit).</p>
<p>Here we will download raw sequencing data directly from NCBI SRA:</p>
<ol>
<li>
<p>Navigate back to the <a href="https://www.ncbi.nlm.nih.gov">NCBI homepage</a>. </p>
</li>
<li>
<p>Select "SRA" from the dropdown box.</p>
<p><img src="../Pictures/SRA1.png" alt="Description1" width="70%"/></p>
</li>
<li>
<p>Type in the project ID (you can also search for specific sequence accessions or even a general search term such as <em>Mycobacterium tuberculosis</em>). Here we will search for the project ID ERP000436.</p>
<p><img src="../Pictures/SRA2.png" alt="Description1" width="70%"/></p>
</li>
<li>
<p>This will pull up all the sequences associated with this project ID. You can search through these individually to look at the data associated with each sequence. To download the sequencing data we will select the "send results to Run selector" button. </p>
<p><img src="../Pictures/SRA3.png" alt="Description1" width="70%"/></p>
</li>
<li>
<p>This will send you to the 'SRA Run Selector' page, which contains information about the project and the sequences contained within. Here we can see we have 358 runs (samples) that totals 208.58GB of data.</p>
<p><img src="../Pictures/SRA4.png" alt="Description1" width="70%"/></p>
</li>
<li>
<p>If you scroll down you will see each sample with the run accession number and all other associated metadata. We can select the first one in this list:</p>
<p><img src="../Pictures/SRA5.png" alt="Description1" width="70%"/></p>
</li>
<li>
<p>This will take us to the sample page. If we click the "FASTA/FASTQ Download" button, this is where we can obtain the sequence data files:</p>
<p><img src="../Pictures/SRA6.png" alt="Description1" width="70%"/></p>
</li>
<li>
<p>We want the raw sequence data for this sample in the FASTQ format, so we can select the "FASTQ" button. This will automatically start a download of the data.</p>
</li>
<p><img src="../Pictures/SRA7.png" alt="Description1" width="70%"/></p>
</ol>
<p>It is important to note that downloading the data from SRA will result in a single FASTQ file, even if the data are paired data that should be separated into 2 files (forward and reverse reads). To obtain the split FASTQ files, you will need to use the SRA-toolkit as descibed above, or download the data from ENA, as per the following exercise.</p>
<p><br></p>
<h3><a href="https://www.ebi.ac.uk/ena/browser/search">European Nucleotide Archive (ENA)</a></h3>
<p>This online database is another good resource for obtaining raw sequencing data but it also contains sequence assemblies and functional annotation. Sequences submitted to SRA will also be (and vice versa). Downloading data directly from ENA can be faster than SRA. </p>
<p>Here we will search for sequences in Project ID ERP000436 in ENA:</p>
<ol>
<li>
<p>Click the link to ENA above.</p>
</li>
<li>
<p>You will be sent to a search page (below). Type "ERP000436" into the search bar.</p>
<p><img src="../Pictures/ENA1.jpeg" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>This will bring up the following page with all results from that ID. Click on the number under the "Study" section.</p>
<p><img src="../Pictures/ENA2.jpeg" alt="Description1" width="70%"/></p>
</li>
<p><br></p>

<p>The project details will be shown as follows:</p>
<p><img src="../Pictures/ENA3.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<li>
<p>Scroll down and you will find the individual samples stored as part of the study. In this project there are 386 <em>M. tuberculosis</em> samples.</p>
<p><img src="../Pictures/ENA4.jpeg" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>Scroll to the right, you will see the options for download. In this study, both paired-end raw sequencing (FASTQ) files are available and an aligned genome file per sample. More on these formats later. These files are available to download through FTP, but some data are available through SRA (Sequence Read Archive) format, a specialized format for NGS data. This is where you can click on a file and download it.</p>
<p><img src="../Pictures/ENA5.jpeg" alt="Description1" width="70%"/></p>
</li>
</ol>
<p><br></p>
<p>You can also batch download files from ENA using FTP through the command line with the program <a href="https://www.gnu.org/software/wget/manual/wget.html">wget</a>. This can be more efficient when downloading large amounts of data. You will need to know the name and path to the files you want to download. These can be found in the report file for each project, found here:</p>
<p><img src="../Pictures/ENA6.jpeg" alt="Description1" width="70%"/></p>
<p><br></p>
<p>This command will just download the two paired-end FASTQ files shown from project ERP000436, but you can develop more complex command line scripts to iterate through many file names in parallel.</p>
<pre><code class="language-bash">wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR036/ERR036185/ERR036185_1.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR036/ERR036185/ERR036185_2.fastq.gz
</code></pre>
<p><br></p>
<h3><a href="https://gisaid.org">Global Initiative on Sharing All Influenza Data (GISAID)</a></h3>
<p>The Global Initiative on Sharing All Influenza Data (GISAID) facilitates the open sharing of genetic and clinical data related to influenza and other emerging viruses. Recently, ithas played a pivotal role in the global response to the COVID-19 pandemic by serving as a key platform for sharing genomic data related to SARS-CoV-2.</p>
<ol>
<li>
<p>Click the link to GISAID above. You will be directed to the following page:</p>
<p><img src="../Pictures/GISAID1.jpeg" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>Login by clicking "login" in the top right, you will be directed here:</p>
<p><img src="../Pictures/GISAID2.jpeg" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>After logging in, you will reach this page. As you can see, there are multiple databases in GISAID. Here, we will look at downloading SARS-CoV-2 assembled genomes. These sequence files are found in the "EpiCoV" section, and sequences can be found through the "Search" option:</p>
<p><img src="../Pictures/GISAID3.jpeg" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>You will be directed here. As you can see there are over 17 million sequences that have been submitted:</p>
<p><img src="../Pictures/GISAID4.png" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>We can choose which sequences to download using the options in the boxes on the page, including filtering by location and lineage. Let's type "Taiwan" into the "Location" box:</p>
<p><img src="../Pictures/GISAID5.png" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>There are over 6,000 SARS-CoV-2 sequences from Taiwan that have been deposited. We can further filter the data by variant. Lets type "B.1.1.7" into the "Lineage" box:</p>
<p><img src="../Pictures/GISAID6.png" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>This takes us to 251 SARS-CoV-2 B.1.1.7 sequences from Taiwan that have been deposited on GISAID. We can also filter to only include complete, high coverage geomes with a complete collection date by selecting the boxes on the right:</p>
<p><img src="../Pictures/GISAID7.png" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>Now we have 246 sequences. Let's download these by selecting all by ticking the box above the sequences on the left, and clicking "Download" at the bottom right:</p>
<p><img src="../Pictures/GISAID8.png" alt="Description1" width="70%"/></p>
</li>
<p><br></p>
<li>
<p>This will now give you the option of which data to download. For the sequence data select "Nucleotide Sequences (FASTA)". Another important dataset is the "Patient status metadata", which will download all metadata associated with the sequences such as collection dates.</p>
<p><img src="../Pictures/GISAID9.png" alt="Description1" width="70%"/></p>
</li>
</ol>
<p><br></p>
<p><em>Note, GISAID will only allow downloads of ~5,000 sequences at the same time, and you must have a login to access the database. Any sequence data obtained from GISAID must be reference appropriately, as detailed <a href="https://gisaid.org/publish/">here</a>.</em></p>
<p><br></p>
<p>Next acitvity: <a href="View_raw_sequence.html">Viewing raw sequence data (FASTQ) files</a></p>
		
	</div>
  <script src="../script.js"></script>
</body>


  
</html>
