<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Obtaining Data</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (Bactdaing)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  
  
  <h2>Quality control (QC) of FASTQ files</h2>
<p>As you can see from the previous activity, viewing and checking FASTQ files in this format is not very useful or efficient. </p>
<p>We can instead use a program called FASTQC to view FASTQ files in a more user-friendly way that allows us to see summary statistics for the sequence data. From this visualization, we can check for any potential issues with these data that may need resolving.</p>
<p>We will be looking at sequence data from four different samples (named Test 1-4) from your dataset to illustrate different potential issues. </p>
<p>These sample data are as follows:</p>
<ul>
<li>
<p><strong>Test1.fastq.gz</strong> – Single-end 50bp sequence data from <em>Mycobacterium tuberculosis</em>, collected from sputum and cultured before DNA library preparation. </p>
</li>
<li>
<p><strong>Test2.fastq.gz</strong> – Single-end 125bp sequence data from <em>Shigella flexneri</em>, collected from sputum and cultured before DNA library preparation.</p>
</li>
<li>
<p><strong>Test3_R1.fastq.gz</strong> and <strong>Test3_R2.fastq.gz</strong> – Paired-end 250bp sequence data from <em>Mycobacterium tuberculosis</em>, collected from sputum and cultured before DNA library preparation.</p>
</li>
<li>
<p><strong>Test4_R1.fastq</strong> and <strong>Test4_R2.fastq</strong> – Paired-end 150bp sequence data from <em>Mycobacterium smegmatis</em> that has been transformed using a plasmid vector.</p>
</li>
</ul>
<p><br></p>
<h4>1. Open file “Test1.fastq.gz” in FASTQ by double-clicking the FASTQC icon and selecting “File&gt;Open…” and finding the FASTQ file in the folder in which your data is stored.</h4>
<p><img src="Pictures/FASTQC1.jpeg" alt="Description" width="60%"/></p>
<p>FastQC performs several analyses, described in modules on left hand side of the main window:</p>
<p><img src="Pictures/FASTQC14.jpeg" alt="Description1" width="60%"/></p>
<p>Normal results are marked with green ticks, slightly abnormal with orange exclamation marks and very unusual with red crosses. 
Note that these results should be taken as guidance in the context of our samples and never as pass/fail results. Nevertheless, they will highlight major problematic aspects. Let us now interpret the output of some of the most important modules.</p>
<p><br></p>
<h4>2. Click on 'Basic Statistics'</h4>
<p><img src="Pictures/FASTQC2.jpeg" alt="Description1" width="60%"/></p>
<p>This module provides overall information including the total number of reads, read length and %GC content.</p>
<p><br></p>
<h4>3. Click on 'Per base sequence quality'</h4>
<p><img src="Pictures/FASTQC3.jpeg" alt="Description1" width="60%"/></p>
<p>This view shows the quality values (y-axis) along the read at each nucleotide position (x-axis). The greater this value, the lower the probability that the corresponding base call is incorrect. At each position a Box plot is drawn where the central read line is the median; the yellow box represents the inter-quartile range (25-75%); the upper and lower whiskers the 10% and 90% points; and the blue line correspond to the mean quality.</p>
<p>As is the case with our sample, call quality degrades towards the 3’ end of the read. This is often the case as in almost every sequencing technology, chemistry deteriorates towards the ends of the sequences and, consequently, the quality values declines in a corresponding manner. Reads can be trimmed or clipped to reduce the number of potential sequencing errors and increase the likelihood that a read can be aligned (Next activity).</p>
<p><br></p>
<h4>4. Click on 'Per tile sequence quality'</h4>
<p><img src="Pictures/FASTQC4.jpeg" alt="Description1" width="60%"/></p>
<p>Flow cells on the sequencer are divided into tiles, this analysis compute the sequence quality per tile to determine if there are any loss of sequence quality associated with any particular flow cell. </p>
<p><br></p>
<h4>5. Click on 'Per sequence quality scores'</h4>
<p><img src="Pictures/FASTQC5.jpeg" alt="Description1" width="60%"/></p>
<p>This analysis computes the mean quality of each read and then displays for all observed values the number of reads having such quality. It is expected that a small subset will have universally poor quality. FastQC outputs a warning if the most frequently observed mean is below 27 (corresponding to a 0.2% error rate) and a failure if is below 20 (1% error rate).</p>
<p><br></p>
<h4>6. Click on 'Per base sequence content'</h4>
<p><img src="Pictures/FASTQC6.jpeg" alt="Description1" width="60%"/></p>
<p>This analysis plots out the proportion of each base position in a file for which each of the four normal DNA base has been called. Note that genomes aren't typically comprised of 25% of each base and therefore the ideal situation would be 4 flat lines with reasonable percentages. Any deviation from a straight line indicates that particular positions within reads have an overrepresentation of particular bases, which would not be expected where the reads will be taken from random places within the genome.  This module issues a warning if the difference between A and T, or G and C is greater than 10% in any position, and a failure if over 20%.</p>
<p><br></p>
<h4>7. Click on 'Per base N content'</h4>
<p><img src="Pictures/FASTQC7.jpeg" alt="Description1" width="60%"/></p>
<p>This module outputs the percentage of N base calls at each position. N is usually called when the sequencer is not confident enough to call a conventional base (A, T, C or G). </p>
<p><br></p>
<h4>8. Click on 'Sequence length distribution'</h4>
<p><img src="Pictures/FASTQC8.jpeg" alt="Description1" width="60%"/></p>
<p>Sequencer produce reads of uniform length. Nevertheless, if reads are trimmed or post-processed due to quality issues we will observe variable read lengths, which may not necessarily affect mapping, as long as reads are still a reasonable length (e.g. 40-50bp or more).</p>
<p><br> </p>
<h4>9. Click on 'Sequence duplication levels'</h4>
<p><img src="Pictures/FASTQC9.jpeg" alt="Description1" width="60%"/></p>
<p>This module computes the degree of duplication of every read in the file by looking for exact matches over the whole length of the rest of reads. A dataset with high depth of coverage, namely an over-represented target genome, will report higher duplication levels than one would expect since multiple reads starting at the same position will be found by chance. Again, this may not be a problem and a threshold can be set at the mapping stag to account for any regions with high read duplication levels.</p>
<p><br></p>
<h4>10. Click on 'Overrepresented sequences'</h4>
<p><img src="Pictures/FASTQC10.jpeg" alt="Description1" width="60%"/></p>
<p>In a normal library, every individual sequence will make up a tiny fraction of the whole set. Therefore, overrepresented reads may be indicators of biologically significant sequences (i.e. gene duplication) or contaminants. All reads representing more than 0.1% of the total amount will be listed. Primers and adaptors are the most common sources of contamination.</p>
<p><br></p>
<p>We can keep a record of the FASTQC results. To create a report simply select File &gt; Save Report from the main menu. This will create a folder containing the images from the analyses, an HTML version of module results and text files containing results numeric data. FASTQC can also be run on the command line. For details on running through the commmand line, click <a href="https://manpages.ubuntu.com/manpages/trusty/man1/fastqc.1.html">here</a>.</p>
<p>We have used the FastQC tool to identify problems in raw sequence data. Nevertheless, bad quality issues does not mean that the data are no good; different issues affect the downstream analysis differently and all data can still be used if we applied the right filtering before processing. For example, de novo assembly may be more difficult with poor quality reads, or where there is a lot of contamination.</p>
<p>If we identify some issues with the sequence data, we can then attempt steps to clean the data.</p>
<h3>Extended exercise:</h3>
<h4>Check the quality of the other sequences using FASTQC (Test2, Test3_1, Test3_2, Test4_1, Test4_3), what potential issues can yoy find in these data?</h4>
<p><br></p>
<p>Next acitvity: <a href="cleaning_fastq.html">Cleaning and filtering FASTQ files</a></p>
  
		
	</div>
</body>


  
</html>