<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Transmission</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (BactDating)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  
<h2>Recombination</h2>
<p>Recombination plays a crucial role in bacterial evolution by enabling the horizontal exchange of genetic material, often leading to the rapid spread of antimicrobial resistance and virulence traits. </p>
<p>Detecting recombination in whole genome sequencing (WGS) data typically involves identifying regions of elevated sequence divergence or patterns inconsistent with inheritance by descent. For example, this can manifest as sections of the genome or mutations that are inconsistent with a constructed phylogeny. Sections of the genome that are likely recombination hotspots can then be masked in alignments to reduce bias caused by these events.</p>
<p>Tools such as Gubbins, ClonalFrameML, and BRATNextGen analyze genome assemblies to detect recombination hotspots and estimate the relative contribution of recombination versus mutation. </p>
<p>When working with genome assemblies, recombination detection often requires accurate alignment and consideration of reference bias, as fragmented or misassembled regions can obscure true recombination signals.</p>
<p>Here we will use ClonalFrameML to conduct a recombination analysis on <em>Klebiella pneumoniae</em> data from Taiwan.</p>
<p>The data we will use for this exercise are:</p>
<ul>
<li>
<p><strong>Klebsiella.fasta</strong> - A FASTA file containing a core genome alignment from 11 de novo assembled Klebsiella pneumoniae strains from Taiwan. </p>
</li>
<li>
<p><strong>Klebsiella.tree</strong> - A newick format tree file constructed using the Klebsiella.fasta file in IQtree.</p>
</li>
</ul>
<p><br></p>
<ol>
<li>Install the required dependencies into your conda environment.</li>
<pre><code class="language-bash">conda install -c conda-forge -c bioconda -c defaults clonalframeml
</code></pre>
<li>Run ClonalFrameML to estimate recombination events that are present in the genome. This will be run with the following command:</li>
<pre><code class="language-bash">ClonalFrameML Klebsiella.tree Klebsiella.tree Klebsiella_recomb
</code></pre>
<p><em>This can take around 15 minutes to run so please put this on before lunch.</em></p>
<h4>After ClonalFrameML has completed, it will have produced the following files:</h4>
<ul>
<li>Klebsiella_recomb.ML_sequence.fasta - Sequence reconstructed by maximum likelihood for all internal nodes of the phylogeny.</li>
<li>Klebsiella_recomb.labelled_tree.newick – Tree with nodes labelled on the tree.</li>
<li>Klebsiella_recomb.em.txt – The point estimates for R/theta, nu, delta and the branch lengths.</li>
<li>Klebsiella_recomb.importation_status.txt – Table of recombination events, one line for each event, columns indicate the branch on which the event was found and the first and last genomic positions affected by the recombination event.</li>
<li>Klebsiella_recomb.position_cross_reference.txt - Comma-separated values indicating for each location in the input sequence file the corresponding position in the ML_sequences.fasta file.</li>
</ul>
<li>
<p>Open the "Klebsiella_recomb.importation_status.txt" file. This will show you estimated positions in the alignment of recombination events and on which branches these fall.</p>
</li>
<li>
<p>We can also use an R script that is provided with ClonalFrameML to view the recombination events along the alignment and where this is on the tree. I have provided the R script - cfml_results.R. You can run the script with the following command:</p>
</li>
<pre><code class="language-bash">Rscript cfml_results.R Klebsiella_recomb
</code></pre>
<li>Open the result PDF file - Klebsiella_recomb.cfml.pdf. Discuss what you think this plot shows.</li>
</ol>
<h4>This analysis will identify potential recombination events in your alignments and produce the plots. If you want to determine in which genes the recombination events are found, you will need to refer to the annotation from the pangenome construction.</h4>
<p><br></p>
<h3>Next activity: <a href="ANI.html">Average nucleotide identity</a></h3>
<p><br></p>
		
	</div>
  <script src="../script.js"></script>
</body>


  
</html>
