<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>View Sequence</title>
  <link rel="stylesheet" href="../style.css">
</head>


<div class="sidebar">
  <div class="sidebar-links">
    <a href="../index.html">Home</a>

    <a href="../pre-workshop_instructions.html">Pre-workhop Instructions</a>

    <div class="subsection">
      <div class="section-subtitle">Lectures</div>
      <a href="../Lectures/Lecture1_WGS.pdf">1: Introduction and Key Concepts</a>
      <a href="../Lectures/Lecture2_Variants.pdf">2: Variant Detection and Phylogenetic Trees</a>
      <a href="../Lectures/Lecture3_Phylodynamics.pdf.pdf">3: Practical Applications of WGS and Phylogenetics</a>
      <a href="../Lectures/Lecture4_Advanced.pdf">4: Advanced Applications of WGS</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 1: WGS Data Analysis</div>
      <a href="../Practicals/Obtaining_data.html">Obtaining sequencing data</a>
      <a href="../Practicals/View_raw_sequence.html">Viewing raw sequencing data</a>
      <a href="../Practicals/Quality_control.html">QC of FASTQ files</a>
      <a href="../Practicals/cleaning_fastq.html">Cleaning and filtering FASTQs</a>
      <a href="../Practicals/mapping.html">Reference mapping/alignment</a>
      <a href="../Practicals/denovo.html">De novo assembly</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 2: Variant Calling and ML Trees</div>
      <a href="../Practicals/VariantCall.html">Variant calling and VCFs</a>
      <a href="../Practicals/Consensus_sequence.html">Building consensus sequences</a>
      <a href="../Practicals/Align_sequence.html">Aligning consensus sequences</a>
      <a href="../Practicals/ML_trees.html">Maximum Likelihood trees</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 3: Timed Phylogenetic Trees</div>
      <a href="../Practicals/Treetime.html">ML timed trees</a>
      <a href="../Practicals/BactDating.html">Bayesian trees (Bactdaing)</a>
      <a href="../Practicals/Beast.html">Bayesian trees (BEAST2)</a>
    </div>

    <div class="subsection">
      <div class="section-subtitle">Practical 4: Transmission and Profiling</div>
      <a href="../Practicals/Identification.html">Identifying lineages and serotypes</a>
      <a href="../Practicals/Transmission.html">Transmission network reconstruction</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 5: Mixed Infection, Recombination and ANI</div>
      <a href="../Practicals/MixInfect.html">Identifying mixed infection</a>
      <a href="../Practicals/Recombination.html">Detecting recombination</a>
      <a href="../Practicals/ANI.html">Average nucleotide identity (ANI)</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 6: Phylogeography and Phylodynamics</div>
      <a href="../Practicals/Ancestral_reconstruction.html">Phylogeography</a>
      <a href="../Practicals/Phylodynamics_beast.html">Inferring population dynamics</a>
    </div>

    
    <div class="subsection">
      <div class="section-subtitle">Practical 7: Fitness and Selection</div>
      <a href="../Practicals/LBI.html">Local branching index (LBI)</a>
      <a href="../Practicals/Homoplasy.html">Detecing homoplasy</a>
      <a href="../Practicals/dnds.html">dN/dS</a>
      <a href="../Practicals/GWAS.html">GWAS</a>
    </div>

    <br>
    <br>
    
  </div>
</div>







<body>
  <div class="content">
  
  
  <h2>Viewing raw sequencing data and understanding the FASTQ format</h2>
<p>We will first open a FASTQ file so you can see how the raw data looks, though you would not be expected to do any manual QC on the sequence data just by viewing it as these files can contain millions of reads.</p>
<p>We can view the FASTQ file in two ways, either through a text viewer or with the command-line in your terminal. If files are in the zipped format (*.fastq.gz) they will need to be unzipped first to view in this format.</p>
<p>Either:
- Open the “Test4_R1.fastq” file in your “Data” folder in your extended text viewer.</p>
<ul>
<li>View the “Test4_R1.fastq” in the command line by entering the following command (note, make sure current working directory is the same folder as the file):</li>
</ul>
<pre><code class="language-bash">head -10 Test4_1.fastq
</code></pre>
<p>The “head’ command will output the first N lines of the file, where N is specified by -N (in our case we have asked for the first 10 lines).</p>
<p><br></p>
<p>The output should look like this:</p>
<p><img src="Pictures/FASTQ_1.png" alt="Description" width="100%"/></p>
<p>If we look more closely at the file, you will see that it comprised of groups of 4 lines of text for each sequenced read: </p>
<p><img src="Pictures/FASTQ_2.png" alt="Description" width="50%"/></p>
<p>These 4 lines consist of the following information:</p>
<ol>
<li>
<p>A sequence identifier with information about the sequencing run and the cluster. The exact contents of this line vary by based on the BCL to FASTQ conversion software used. If the sequence data are paired, there will be a \1 or \2 at the end of this line to denote the forward or reverse read (all reads included will have a forward and reverse entry, called a ‘mate pair’)</p>
</li>
<li>
<p>The sequence (the base calls; A, C, T, G, and N if base cannot be confidently called).</p>
</li>
<li>
<p>A separator (+) between the sequence and quality line.</p>
</li>
<li>
<p>The base call quality scores. These are Phred +33 encoded, using ASCII characters to represent the numerical quality scores. Further information can be found here: https://help.basespace.illumina.com/files-used-by-basespace/quality-scores</p>
</li>
</ol>
<p><br>
The next step is to assess the quality of these raw sequences and perform additional steps to resolve any potential issues.</p>
<p><br> <strong>Next activity:</strong> <a href="Quality_control.html">Quality control (QC) of FASTQ files</a></p>
  
		
	</div>
</body>


  
</html>